RecoEcal/EgammaCoreTools/src/EcalClusterLazyTools.cc

```diff
@@ -257,11 +257,11 @@ float EcalClusterLazyToolsBase::eseffsirir(const reco::SuperCluster &cluster)
   if (!(fabs(cluster.eta()) > 1.6 && fabs(cluster.eta()) < 3.)) return 0.;
 
   const CaloSubdetectorGeometry *geometryES = geometry_->getSubdetectorGeometry(DetId::Ecal, EcalPreshower);
-  CaloSubdetectorTopology *topology_p = 0;
-  if (geometryES) topology_p = new EcalPreshowerTopology(geometry_);
+  std::unique_ptr<CaloSubdetectorTopology> topology_p;
+  if (geometryES) topology_p.reset(new EcalPreshowerTopology(geometry_));
```

Comment:
Ok, I see, `getSubdetectorGeometry()` may return a null-pointer if something is wrong, so this check does probably make sense.